### PR TITLE
Add docstrings for Validator and its save_expectation_suite and validate methods

### PR DESCRIPTION
**Pull request by wookasz**
Changes proposed in this pull request:
- Documentation for `Validator` class and its `save_expectation_suite` and `validate` methods

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

### CODE REVIEW

File: great_expectations/validator/validator.py [hunk 1]
```diff
@@ -1579,81 +1574,32 @@ def validate(  # noqa: C901 - complexity 31
         run_name: Optional[str] = None,
         run_time: Optional[str] = None,
         checkpoint_name: Optional[str] = None,
-    ) -> Union[ExpectationValidationResult, ExpectationSuiteValidationResult]:
+    ) -> Union[
+        ExpectationValidationResult, ExpectationSuiteValidationResult
+    ]:  # noqa: C901 - 31
```
Comment (Kilo59) [inline great_expectations/validator/validator.py]:
```suggestion
    ]:
```
You don't need it here.

File: great_expectations/validator/validator.py [hunk 2]
```diff
@@ -1565,10 +1554,16 @@ def save_expectation_suite(
             )
 
     # TODO: <Alex>Should "include_config" also be an argument of this method?</Alex>
+    @public_api
```
Comment (Kilo59) [inline great_expectations/validator/validator.py]:
```suggestion
    @public_api   # noqa: C901
```

File: great_expectations/validator/validator.py [hunk 3]
```diff
@@ -1565,10 +1554,16 @@ def save_expectation_suite(
...
+    @deprecated_argument(
```
Comment (Kilo59) [inline great_expectations/validator/validator.py:1555]:
```suggestion
    @deprecated_argument(  # noqa: C901
```
It's possible you'll need it here too but hopefully not 😬

File: great_expectations/validator/validator.py [hunk 4]
```diff
@@ -1516,32 +1517,20 @@ def save_expectation_suite(
...
-               discard_include_config_kwargs (boolean): \
-                   If True, the :ref:`include_config` attribute for each expectation is not written to the JSON config \
-                   file.
-               discard_catch_exceptions_kwargs (boolean): \
-                   If True, the :ref:`catch_exceptions` attribute for each expectation is not written to the JSON \
-                   config file.
-               suppress_warnings (boolean): \
-                    If True, all warnings raised by Great Expectations, as a result of dropped expectations, are \
-                    suppressed.
+        """Write the configuration to the Expectation Store associated with the Validator's Data Context.
```
Comment (anthonyburdi) [inline great_expectations/validator/validator.py]:
```suggestion
        """Write the Expectation Suite (e.g. from interactive evaluation) to the Expectation Store associated with the Validator's Data Context.
```

**Review by anthonyburdi — APPROVED** [review]
A thing of beauty. Thank you for describing the various parameters and configuration options! Just a tiny suggestion.

**Pull request merged**